### PR TITLE
[ramda] Add type parameters for cond

### PR DESCRIPTION
**Pull request by alexandradeas**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by leighman** [comment]
Does the test taken from the ramda docs still work?

Ramda also supports eg.
```
const fn = R.cond([
  [(n, m) => n + m > 3,   R.always('Hi')],
  [R.T,   R.always('Blah')],
]);
fn(2, 3)
```
which is not supported by your types, I think.

**Review by KiaraGrouwstra — APPROVED** [review]

**Comment by amcasey** [comment]
@alex-deas any comment on @leighman's test case?

**Comment by amcasey** [comment]
Given that this seems to break a valid, if untested, use case, I'm inclined to close the PR.  Any objections?

**Review by amcasey — CHANGES_REQUESTED** [review]
Flipping my status for tracking purposes.

File: types/ramda/index.d.ts [hunk 1]
```diff
@@ -777,6 +777,7 @@ declare namespace R {
          * matches, fn returns undefined.
          */
         cond(fns: ReadonlyArray<[Pred, (...a: any[]) => any]>): (...a: any[]) => any;
+        cond<A, B>(fns: ReadonlyArray<[SafePred<A>, (...a: A[]) => B]>): (...a: A[]) => B;
```
Comment (amcasey) [inline types/ramda/index.d.ts]: Unless I'm mistaken, this overload will only be used if explicit type arguments are provided.  Was that your expectation?

Comment (alex-deas) [reply types/ramda/index.d.ts]: It was, unless there's another method I'm not aware of there's no way for TS to reliably infer the type here without breaking existing code. Else `tsc` infers A & B as `never`.

Opting in to the type allows people to type the cond if they want or preserve their existing assumptions about types.

Comment (amcasey) [reply types/ramda/index.d.ts]: It seems reasonable - I just didn't want it to be a nasty surprise. :wink:

**Comment by amcasey** [comment]
@tycho01 @leighman All good?

**Comment by amcasey** [comment]
@leighman @alex-deas, I was actually going to suggest the same - better to keep the existing tests (and add new ones) if retaining the existing signature as-is.

**Comment by alex-deas** [comment]
I have no problem with adding in the original tests but do they add much which isn't covered in the current test cases?

**Comment by amcasey** [comment]
If the same helpers are covered elsewhere, I'm happy to merge.

**Review by amcasey — APPROVED** [review]

**Comment by leighman** [comment]
They test the interaction with other ramda functions which seems to be where I meet problems personally.

**Pull request merged**